File: Best Time to Buy and Sell Stock.py
class Solution(object):
    def maxProfit(self, prices):
        maxProfit = 0
        if prices:
            minPrice = prices[0]
        for i in prices:
            if i < minPrice:
                minPrice = i
            else:
                if i-minPrice > maxProfit:
                    maxProfit = i-minPrice
        return maxProfit

File: Best Time to Buy and Sell Stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int minTillNow = INT_MAX, maxProfit = 0;
        for(int i:prices){
            //assume u will sell at this day with value i
            //and u already bought the stock at min cost day 
            maxProfit = max(maxProfit, i-minTillNow);
            minTillNow = min(minTillNow, i);
        }
        return maxProfit;
    }
};
